astropy/modeling/tests/test_quantities_evaluation.py

```diff
@@ -0,0 +1,538 @@
+            return 2 * x
+
+    m = TestModel()
+    assert_quantity_allclose(m(3 * u.nm), 6 * u.nm)
+
+    # We now check it works with frequency. Note that we should make sure
+    # we evaluate the model outside the context manager to make sure that we
+    # are picking up the equivalency from input_equivalencies. The context
+    # manager is only there because assert_quantity_allclose doesn't support
+    # equivalencies.
+    result = m(400 * u.GHz)
+    with u.set_enabled_equivalencies(u.spectral()):
+        assert_quantity_allclose(result, 800 * u.GHz)
+
+    # And check that changing the equivalencies on-the-fly works
+    m.input_equivalencies = None
+    with pytest.raises(UnitsError) as exc:
+        m(400 * u.GHz)
+    assert exc.value.args[0] == ("Units of input 'x', GHz (frequency), could "
+                                 "not be converted to required input units of "
+                                 "m (length)")
+
+
+@pytest.mark.xfail
+def test_output_units_equivalencies():
+    """
+    Test that output_equivalencies is recognized
+    """
+
+    class TestModel(Fittable1DModel):
+        output_units = u.m
+        output_equivalencies = u.spectral()
+        @staticmethod
+        def evaluate(x):
+            return 2 * x
+
+    m = TestModel()
+    assert_quantity_allclose(m(3 * u.nm), 6 * u.nm)
+
+    # We now check it works with frequency. As before, we need to evaluate the
+    # model outside the context manager.
+    result = m(400 * u.GHz)
+
+    # The output units should actually be meters since that is what was
+    # requested - we're just checking that no error was raised above.
+    assert result.unit is u.m
+
+    # And we now check the actual accuracy of the result to be sure
+    with u.set_enabled_equivalencies(u.spectral()):
+        assert_quantity_allclose(result, 800 * u.GHz)
+
+    # And check that changing the equivalencies on-the-fly works
+    m.output_equivalencies = None
+    with pytest.raises(UnitsError) as exc:
+        m(400 * u.GHz)
+    assert exc.value.args[0] == ("Units of output 'y', GHz (frequency), could "
+                                 "not be converted to required output units of "
+                                 "m (length)")
+
+
+@pytest.mark.xfail
+def test_output_units_equivalencies_with_parameters():
+    """
+    Test using equivalencies that require parameters that are input values.
+    """
+
+    # We choose a tricky example, brightness_temperature, which has two
+    # parameters: the beam size, and the dispersion coordination, which is the x
+    # coordinate in the model.
+
+    # TODO: the API below is just an example of one way to deal with it. We
+    # basically set the equivalency to a tuple of the form:
+    #
+    # g.output_equivalencies = equivalency, [args, [kwargs]]
+    #
+    # where args and kwargs are both optional (as indicated by the []). In
+    # addition to values, strings indicating parameters or the input value
+    # (e.g. 'x') can be used.
```

Comment:
Some discussion of alternate syntax is [here](https://github.com/astropy/astropy/pull/4615/files#r53465486), specifically using a dict instead of a tuple to specify the output equivalencies